### DISCUSSION TITLE
[E109] Model for component 'textcat' not initialized. Did you forget to load a model, or forget to call begin_training()?

### DISCUSSION THREAD

**Discussion by Y4rd13** [explosion/spaCy#5997] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour
1. Go to google colab
2. write:
```python
import spacy
from spacy.pipeline import TextCategorizer

spNLP = spacy.load('en_core_web_sm')
textcat = spNLP.create_pipe(name='textcat', config={'exclusive_classes': True})
textcat = TextCategorizer(spNLP.vocab)

text = spNLP('this is a sentence.')
processed = textcat(text)
```
3. see issue

## Your Environment
google cloud default enviroment

## **Traceback**
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-42-d61740aa473b> in <module>()
      1 text = spNLP('this is a sentence.')
----> 2 processed = textcat(text)

pipes.pyx in spacy.pipeline.pipes.Pipe.__call__()

pipes.pyx in spacy.pipeline.pipes.Pipe.require_model()

ValueError: [E109] Model for component 'textcat' not initialized. Did you forget to load a model, or forget to call begin_training()?
```

**Comment by svlandeg** [comment, marked as answer]
It looks like your component is not yet initialized (as the error message points out). You'll need to call
`spNLP.begin_training()`. In fact, after that you'll also want to train your new component, because right now the textcat will not be able to predict anything useful. There's more information in the docs [here](https://spacy.io/usage/training#training-simple-style), and on the rest of that page. I would suggest to go through some of those examples and copy/paste the code, run it and edit it, to get more familiar with how spaCy works! Have fun ;-)
